sdk/keyvault/azure-security-test-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultCertificatesTest.java

```diff
@@ -0,0 +1,113 @@
+        + "frDEVOO9gwZbz7xppyqutjxjllw8CCgjFdfK02btz56CGgh3X25ZZtzPbuMZJM0j"
+        + "o4mVEdaFNJ0eUeMppS0DcbbuTWCF7Jf1gvr8GVqx+E0IJUFkE+D4kdTbnJSaeK0A"
+        + "KEt94z88MPX18h8ud14uRVmUCYVZrZeswdE2tO1BpazrXELHuXCtrjGxsDDjDzeP"
+        + "98aFI9kblkqoJS4TsmloLEjwZLm80cyJDEmpXXMtR7C0FFXFI1BAtIa4mxSgBLsT"
+        + "L4GVPEGNANR8COYkHQIDAQABo4GjMIGgMA4GA1UdDwEB/wQEAwIFoDAJBgNVHRME"
+        + "AjAAMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAkBgNVHREEHTAbghlo"
+        + "dW5kcmVkLXllYXJzLmV4YW1wbGUuY29tMB8GA1UdIwQYMBaAFOGTt4H3ho30O4e+"
+        + "hebwJjm2VMvIMB0GA1UdDgQWBBThk7eB94aN9DuHvoXm8CY5tlTLyDANBgkqhkiG"
+        + "9w0BAQsFAAOCAQEAGp8mCioVCmM+kZv6r+K2j2uog1k4HBwN1NfRoSsibDB8+QXF"
+        + "bmNf3M0imiuR/KJgODyuROwaa/AalxNFMOP8XTL2YmP7XsddBs9ONHHQXKjY/Ojl"
+        + "PsIPR7vZjwYPfEB+XEKl2fOIxDQQ921POBV7M6DdTC49T5X+FsLR1AIIfinVetT9"
+        + "QmNuvzulBX0T0rea/qpcPK4HTj7ToyImOaf8sXRv2s2ODLUrKWu5hhTNH2l6RIkQ"
+        + "U/aIAdQRfDaSE9jhtcVu5d5kCgBs7nz5AzeCisDPo5zIt4Mxej3iVaAJ79oEbHOE"
+        + "p192KLXLV/pscA4Wgb+PJ8AAEa5B6xq8p9JO+Q==";
+
+    @BeforeAll
+    public static void setEnvironmentProperty() {
+        PropertyConvertorUtils.putEnvironmentPropertyToSystemProperty(
+            Arrays.asList("AZURE_KEYVAULT_URI",
+                "AZURE_KEYVAULT_TENANT_ID",
+                "AZURE_KEYVAULT_CLIENT_ID",
+                "AZURE_KEYVAULT_CLIENT_SECRET")
+        );
+        Security.insertProviderAt(new KeyVaultJcaProvider(), 1);
+    }
+
+    @Test
+    public void testGetKeyStore() throws Exception {
+        KeyStore keyStore = PropertyConvertorUtils.getKeyVaultKeyStore();
+        assertNotNull(keyStore.getCertificate("myalias"));
+        assertTrue(keyStore.containsAlias("myalias"));
+        X509Certificate certificate = getTestCertificate();
+
+        keyStore.setCertificateEntry("setcert", certificate);
+        assertNotNull(keyStore.getCertificateAlias(certificate), "setcert");
+    }
+
+    private X509Certificate getTestCertificate() {
+        X509Certificate certificate;
+
+        try {
+            byte[] certificateBytes = Base64.getDecoder().decode(TEST_CERTIFICATE);
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certificateBytes));
+        } catch (CertificateException e) {
+            throw new ProviderException(e);
+        }
+        return certificate;
+    }
+
+
+
+    @Test
+    public void testCertificatesRefreshInterval() throws Exception {
+        System.setProperty("azure.keyvault.jca.certificates-refresh-interval", "15000");
+        KeyStore keyStore = PropertyConvertorUtils.getKeyVaultKeyStore();
+        X509Certificate certificate = getTestCertificate();
+
+        assertNotNull(keyStore.getCertificate("myalias"));
+        keyStore.deleteEntry("myalias");
+        keyStore.setCertificateEntry("myalias", certificate);
```

Comment:
`"azure.keyvault.jca.certificates-refresh-interval"` is used in `KeyVaultCertificates`, but `keyStore.setCertificateEntry()` will affect `ClasspathCertificates`. So I'm  confused about this line.

Reply:
According to the logic in the original jca, when loading, if there is an alias certificate in the keyvault, the local certificate with the same alias will not be loaded into the sideload. ([related code](https://github.com/Azure/azure-sdk-for-java/blob/8d398b00376e10e442ccca7ef1fd00745ecb6d0e/sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java#L349) & [set alias code](https://github.com/Azure/azure-sdk-for-java/blob/8d398b00376e10e442ccca7ef1fd00745ecb6d0e/sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java#L249-#L257))
So I think the priority of the certificate is keyvault > sideload(classpath).
Because both KeyVaultCertificates and ClasspathCertificates contain the certificate whose alias is `myalias`, I think we should take out the certificate in KeyVaultCertificates (because I think the same alias can only correspond to one certificate, maybe my understanding is wrong?)